controllers/server_controller.go

```diff
@@ -266,6 +286,57 @@ func (r *ServerReconciler) createExperiment(ctx context.Context, log logr.Logger
 	return nil, nil
 }
 
+func (r *ServerReconciler) createApplication(ctx context.Context, exp *optimizev1beta2.Experiment) error {
+	appName := applications.ApplicationName(exp.GetLabels()[optimizeappsv1alpha1.LabelApplication])
+	scnName := applications.ScenarioName(exp.GetLabels()[optimizeappsv1alpha1.LabelScenario])
+
+	if appName == "" || r.ApplicationsAPI == nil {
+		return nil
+	}
+
+	// Check to see if the application exists
+	app, err := r.ApplicationsAPI.GetApplicationByName(ctx, appName)
+	if controller.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	// A non-nil error here means it was not found, try creating it
+	if err != nil {
+		if _, err := r.ApplicationsAPI.UpsertApplicationByName(ctx, appName, applications.Application{}); err != nil {
```

Comment:
should we add in namespace name here from experiment?

Reply:
I don't think the experiment's name doesn't necessarily correspond to the application resources: even if they did I'm not sure we want to make the association via this path to avoid confusion with applications coming from other workflows.